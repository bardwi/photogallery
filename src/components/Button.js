import React from "react";
import "./Button.scss";

const Button = (props) => {
    return(
        <button  onClick={() => props.addFavorites(props.id)}
              className={props.isFavorite?"btn-favourite":"btn-not-favourite"}
            >
            Favourite
        </button>
    )
}

export default Button;