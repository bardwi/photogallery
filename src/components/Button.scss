@import '../variables';

.btn-favourite {
    background-color: $color-text-white;
    color: $color-text-black;
    border: 1px solid $color-text-white;
    margin: 4px 2px;
    padding: 15px 32px;
    cursor: pointer;
    font-size: 1rem;
    border-radius: 30px;
    mix-blend-mode: screen;
  }
  
  .btn-not-favourite {
    background-color: transparent;
    color: $color-text-white;
    border: 1px solid $color-text-white;
    margin: 4px 2px;
    padding: 15px 32px;
    cursor: pointer;
    font-size: 1rem;
    border-radius: 30px;
  }