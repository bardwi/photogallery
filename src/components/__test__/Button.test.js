import React from "react";

import {render, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom'
import Button from '../Button';


describe("Button Component" ,() => {
    const props = {
        addFavorites: jest.fn(),
        text: 'Favourite',
    }
    test("renders component", ()=> {
        const {button} = render(<Button />);
        expect(button).toMatchSnapshot();
    })

    test("Calls the on Click", ()=> {
    
    const { addFavorites, text } = props;
    const {queryByText} = render(
    <Button {...props} addFavorites={addFavorites}/>);
    fireEvent.click(queryByText(text));
    expect(addFavorites).toHaveBeenCalled()
    })

})