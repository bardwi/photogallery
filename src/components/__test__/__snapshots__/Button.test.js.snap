// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders component 1`] = `undefined`;
